### ISSUE TITLE
unprompted crash

### ISSUE THREAD

**Issue by ThePepster** [spyder-ide/spyder#20803]
### What steps will reproduce the problem?

1. Training a Keras model outputting to the console.
2. Window was minimized an unattended.
3. Came back and Spyder has exception and is not receiving updates from the consile. Training is still ongoing in the background but has no visible output.

### Traceback
```python-traceback
Traceback (most recent call last):
  File "c:\users\pep\desktop\wpy64-3870\python-3.8.7.amd64\lib\site-packages\spyder\plugins\application\container.py", line 331, in _check_updates_ready
    msg = header + content + footer
UnboundLocalError: local variable 'content' referenced before assignment
```

## Versions

* Spyder version: 5.4.2  (pip)
* Python version: 3.8.7 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.8
* Operating System: Windows 10

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.1 (OK)
chardet >=2.0.0               :  5.1.0 (OK)
cloudpickle >=0.5.0           :  2.2.1 (OK)
cookiecutter >=1.6.0          :  2.1.1 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.31.1;<9.0.0       :  8.9.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.17.2 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  4.17.3 (OK)
keyring >=17.0.0              :  23.13.1 (OK)
nbconvert >=4.0               :  7.2.9 (OK)
numpydoc >=0.6.0              :  1.5.0 (OK)
paramiko >=2.4.0              :  3.0.0 (OK)
parso >=0.7.0;<0.9.0          :  0.7.1 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.4 (OK)
pygments >=2.0                :  2.14.0 (OK)
pylint >=2.5.0;<3.0           :  2.15.10 (OK)
pylint_venv >=2.1.1           :  2.3.0 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.7.1;<1.8.0          :  1.7.1 (OK)
pylsp_black >=1.2.0           :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0     :  3.0.3 (OK)
qstylizer >=0.2.2             :  0.2.2 (OK)
qtawesome >=1.2.1             :  1.2.2 (OK)
qtconsole >=5.4.0;<5.5.0      :  5.4.0 (OK)
qtpy >=2.1.0                  :  2.3.0 (OK)
rtree >=0.9.7                 :  1.0.1 (OK)
setuptools >=49.6.0           :  67.0.0 (OK)
sphinx >=0.6.6                :  6.1.3 (OK)
spyder_kernels >=2.4.2;<2.5.0 :  2.4.2 (OK)
textdistance >=4.2.0          :  4.5.0 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.2.1 (OK)
zmq >=22.1.0                  :  25.0.0 (OK)

# Optional:
cython >=0.21                 :  None (NOK)
matplotlib >=3.0.0            :  3.6.3 (OK)
numpy >=1.7                   :  1.24.1 (OK)
pandas >=1.1.1                :  1.5.3 (OK)
scipy >=0.17.0                :  None (NOK)
sympy >=0.7.3                 :  None (NOK)
```

### Environment

<details><summary>Environment</summary>

```
Package                       Version
----------------------------- -------------
absl-py                       1.4.0
alabaster                     0.7.13
appdirs                       1.4.4
arrow                         1.2.3
astroid                       2.13.3
asttokens                     2.2.1
astunparse                    1.6.3
async-generator               1.10
atomicwrites                  1.4.1
attrs                         22.2.0
autopep8                      1.6.0
Babel                         2.11.0
backcall                      0.2.0
bcrypt                        4.0.1
beautifulsoup4                4.11.1
binaryornot                   0.4.4
black                         22.12.0
bleach                        6.0.0
cachetools                    5.3.0
certifi                       2022.12.7
cffi                          1.15.1
chardet                       5.1.0
charset-normalizer            2.0.12
clang                         14.0
click                         8.1.3
cloudpickle                   2.2.1
colorama                      0.4.6
comm                          0.1.2
contourpy                     1.0.7
cookiecutter                  2.1.1
cryptography                  39.0.0
cycler                        0.11.0
debugpy                       1.6.6
decorator                     5.1.1
defusedxml                    0.7.1
diff-match-patch              20200713
dill                          0.3.6
dm-tree                       0.1.8
docstring-to-markdown         0.11
docutils                      0.19
entrypoints                   0.4
etils                         1.0.0
exchange-calendars            4.2.5
executing                     1.2.0
fastjsonschema                2.16.2
flake8                        6.0.0
flatbuffers                   23.1.21
fonttools                     4.38.0
future                        0.18.3
gast                          0.4.0
google-auth                   2.16.0
google-auth-oauthlib          0.4.6
google-pasta                  0.2.0
googleapis-common-protos      1.58.0
grpcio                        1.51.1
h5py                          3.8.0
helpdev                       0.7.1
idna                          3.4
imagesize                     1.4.1
importlib-metadata            6.0.0
importlib-resources           5.10.2
inflection                    0.5.1
int-date                      0.2.0
intervaltree                  3.1.0
ipykernel                     6.21.0
ipython                       8.9.0
ipython-genutils              0.2.0
isort                         5.12.0
jaraco.classes                3.2.3
jedi                          0.17.2
jellyfish                     0.9.0
Jinja2                        3.1.2
jinja2-time                   0.2.0
joblib                        1.2.0
jsonschema                    4.17.3
jupyter_client                7.4.9
jupyter_core                  5.2.0
jupyterlab-pygments           0.2.2
keras                         2.10.0
Keras-Preprocessing           1.1.2
keyring                       23.13.1
kiwisolver                    1.4.4
korean-lunar-calendar         0.3.1
lazy-object-proxy             1.9.0
libclang                      15.0.6.1
lxml                          4.9.2
Markdown                      3.4.1
MarkupSafe                    2.1.2
matplotlib                    3.6.3
matplotlib-inline             0.1.6
mccabe                        0.7.0
mistune                       2.0.4
more-itertools                9.0.0
msvc-runtime                  14.34.31931
mypy-extensions               0.4.3
nbclient                      0.7.2
nbconvert                     7.2.9
nbformat                      5.7.3
nest-asyncio                  1.5.6
numpy                         1.24.1
numpydoc                      1.5.0
oauthlib                      3.2.2
opt-einsum                    3.3.0
packaging                     23.0
pandas                        1.5.3
pandas-datareader             0.10.0
pandas-market-calendars       4.1.3
pandocfilters                 1.5.0
paramiko                      3.0.0
parso                         0.7.1
pathspec                      0.11.0
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        9.4.0
pip                           23.0
pkgutil_resolve_name          1.3.10
platformdirs                  2.6.2
pluggy                        1.0.0
poyo                          0.5.0
promise                       2.3
prompt-toolkit                3.0.36
protobuf                      3.19.6
psutil                        5.9.4
ptyprocess                    0.7.0
pure-eval                     0.2.2
pyasn1                        0.4.8
pyasn1-modules                0.2.8
pycodestyle                   2.10.0
pycparser                     2.21
pydocstyle                    6.2.3
pyflakes                      3.0.1
Pygments                      2.14.0
pylint                        2.15.10
pylint-venv                   2.3.0
pyls-black                    0.4.7
pyls-spyder                   0.4.0
pyluach                       2.0.2
PyNaCl                        1.5.0
pyparsing                     3.0.9
PyQt5                         5.15.8
PyQt5-Qt5                     5.15.2
PyQt5-sip                     12.11.1
PyQtWebEngine                 5.15.6
PyQtWebEngine-Qt5             5.15.2
pyrsistent                    0.19.3
python-dateutil               2.8.2
python-jsonrpc-server         0.4.0
python-language-server        0.36.2
python-lsp-black              1.2.1
python-lsp-jsonrpc            1.0.0
python-lsp-server             1.7.1
python-slugify                8.0.0
pytoolconfig                  1.2.5
pytz                          2022.7.1
pywin32                       305
pywin32-ctypes                0.2.0
PyYAML                        6.0
pyzmq                         25.0.0
QDarkStyle                    3.0.3
qstylizer                     0.2.2
QtAwesome                     1.2.2
qtconsole                     5.4.0
QtPy                          2.3.0
regex                         2022.10.31
requests                      2.28.2
requests-oauthlib             1.3.1
rope                          1.7.0
rsa                           4.9
Rtree                         1.0.1
setuptools                    67.0.0
six                           1.16.0
snowballstemmer               2.2.0
sortedcontainers              2.4.0
soupsieve                     2.3.2.post1
Sphinx                        6.1.3
sphinxcontrib-applehelp       1.0.4
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
spyder                        5.4.2
spyder-kernels                2.4.2
sqlite-bro                    0.12.2
stack-data                    0.6.2
stockstats                    0.5.1
tensorboard                   2.10.1
tensorboard-data-server       0.6.1
tensorboard-plugin-wit        1.8.1
tensorflow                    2.10.1
tensorflow-datasets           4.8.2
tensorflow-estimator          2.10.0
tensorflow-hub                0.12.0
tensorflow-intel              2.11.0
tensorflow-io-gcs-filesystem  0.30.0
tensorflow-metadata           1.12.0
tensorflow-text               2.10.0
termcolor                     2.2.0
testpath                      0.6.0
text-unidecode                1.3
textdistance                  4.5.0
three-merge                   0.1.1
tinycss2                      1.2.1
toml                          0.10.2
tomli                         2.0.1
tomlkit                       0.11.6
toolz                         0.12.0
tornado                       6.2
tqdm                          4.64.1
traitlets                     5.9.0
typed-ast                     1.5.4
typing_extensions             4.4.0
ujson                         5.7.0
urllib3                       1.26.14
watchdog                      2.2.1
wcwidth                       0.2.6
webencodings                  0.5.1
Werkzeug                      2.2.2
whatthepatch                  1.0.4
wheel                         0.38.4
winpython                     3.0.202011219
wrapt                         1.14.1
xlwings                       0.29.0
yapf                          0.32.0
zipp                          3.12.0

```
</details>

**Comment by dalthviz** [comment]
Hi @ThePepster thank you for the feedback! This is a duplicate of #20406 To prevent this you will need to either deactivate the check for updates at startup option or update Spyder to the latest release (5.4.3). The fix for this was released with Spyder 5.4.3 so hopefully if you update to 5.4.3 then when we release a new Spyder version you will not get this error again.

Let us know if the info above helps!

**Comment by dalthviz** [comment]
Closing as a duplicate of https://github.com/spyder-ide/spyder/issues/20406
